master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md

```diff
@@ -22,25 +25,79 @@ You must have a cluster which meets the following requirements:
 
+> Note:  If you are upgrading from Calico v2.1, the cluster-cidr selected for your controller manager should remain
+> unchanged from the v2.1 install (the v2.1 manifests default to `10.244.0.0/16`).
+
 ## Installation
 
-To install Calico, ensure you have a cluster which meets the above requirements and run one of the following commands based on your Kubernetes version:
+This document describes three installation options for Calico using Kubernetes API as the datastore:
+
+-  Calico policy with Calico networking
+-  Calico policy-only
+   -  with user-supplied networking
+   -  with flannel networking
+
+Ensure you have a cluster which meets the above requirements.  There may be additional requirements based on the installation option you choose.
+
+> Note:  There is currently no upgrade path to switch between different installation options.  Therefore, 
+> if you are upgrading from Calico v2.1, use the [Calico in policy-only mode](#calico-in-policy-only-mode) installation instructions 
+> to upgrade Calico policy-only leaving the networking solution unchanged.
+
+### Calico policy with Calico networking (Beta)
+
+With Kubernetes as the Calico datastore, Calico has Beta support for Calico networking.  This provides BGP-based
+networking with a full node-to-node mesh.  It is not currently possible to configure the Calico BGP network to peer with
+other routers - a future release will provide this feature, and to provide feature parity with the [Kubeadm Hosted Install](../kubeadm)
+which uses a separate etcd cluster for Calico.
+
+To install Calico with Calico networking, run one of the following commands based on your Kubernetes version:
 
 For Kubernetes 1.6 clusters:
 
 ```
-kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/1.6/calico.yaml
+kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico-networking/1.6/calico.yaml
 ```
 
->[Click here to view the above yaml directly.](1.6/calico.yaml)
+>[Click here to view the above yaml directly.](calico-networking/1.6/calico.yaml)
 
 For Kubernetes 1.5 clusters:
 
 ```
-kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico.yaml
+kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico-networking/1.5/calico.yaml
 ```
 
->[Click here to view the above yaml directly.](calico.yaml)
+>[Click here to view the above yaml directly.](calico-networking/1.5/calico.yaml)
+
+### Calico policy-only with user-supplied networking
+
+If you run Calico in policy-only mode it is necessary to configure your network to route pod traffic based on pod 
+CIDR allocations, either through static routes, a Kubernetes cloud-provider integration, or flannel.
```

Comment:
Did you purposefully include flannel here even though there is a dedicated section for it?